### PR TITLE
Homework 5.3

### PR DESCRIPTION
**Pull request by SorokinTim**

### CODE REVIEW

File: src/api/groups/groups.ts [hunk 1]
```diff
@@ -1,59 +1,84 @@
...
-    }
+        if (!groups.length) {
+            next(ApiError.dataIsNotAnyMatch());
+            return;
+        }
 
-    res.send(groups);
+        res.send(groups);
+    } catch (error: any) {
+        unknownErrorLoggerMiddleware(error, req, res, next);
```
Comment (potsiahevich-dmitry) [inline src/api/groups/groups.ts:23]: middlewares should be used as middleware. To run the next middleware in the chain you have to call next method

File: src/index.ts [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
+app.all('*', (req, res, next) => {
+    next(ApiError.dataNotFound());
+});
+
+app.use(unhandledErrorsMiddleware);
+
+process.on('uncaughtException', reason => {
+    winstonLoggerMiddleware.error(reason.message);
+    process.exit(1);
 });
```
Comment (potsiahevich-dmitry) [inline src/index.ts:26]: The 'Add Unhandled promiserejection listener to log errors.' is missing

**Review by potsiahevich-dmitry — CHANGES_REQUESTED** [review]

File: src/api/groups/groups.ts (hunk 1 above)
Comment (SorokinTim) [reply src/api/groups/groups.ts]: Thanks, fixed into 03bbdce80f275f0e8c7d776ab1e64d14cdf68ade commit

File: src/index.ts (hunk 2 above)
Comment (SorokinTim) [reply src/index.ts:29]: Thanks, fixed into 797af15e4006fb3c6799998ba46a64e563cee329 commit

**Review by potsiahevich-dmitry — APPROVED** [review]

**Pull request merged**